### PR TITLE
DataStore

### LINKED ISSUE #41: Implement test data version of Login Storage API

### PR DESCRIPTION
**Pull request by linuxwolf**
Connected to #40 
Resolves #41 

Flux Store to wrap LoginStorage.  Initial implementation uses dummy test data.

### CODE REVIEW

File: app/src/main/java/mozilla/lockbox/store/DataStore.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                backend.lock().whenComplete {
+                    doLock()
+                }
+            }
+        }
+    }
+
+    // backend management
+    private fun setupBackend(): LoginsStorage {
+        val testdata = List<ServerPassword>(10) { createTestItem(it) }
```
Comment (sashei) [inline app/src/main/java/mozilla/lockbox/store/DataStore.kt]: This seems like a good opportunity to use a Factory.

Comment (linuxwolf) [reply app/src/main/java/mozilla/lockbox/store/DataStore.kt]: 🤔  ... 👍

**Comment by devinreams** [comment]
>w: /bitrise/src/app/src/main/java/mozilla/lockbox/presenter/SettingPresenter.kt: (9, 24): Parameter 'view' is never used

This will be the case and keep erroring for a little bit... we can suppress it for now until it comes in.

**Comment by linuxwolf** [comment]
> > w: /bitrise/src/app/src/main/java/mozilla/lockbox/presenter/SettingPresenter.kt: (9, 24): Parameter 'view' is never used
> 
> This will be the case and keep erroring for a little bit... we can suppress it for now until it comes in.

That's not the line triggering the failures; there's a bunch of linting errors in `FxALoginFragment.kt` and `ItemListFragment.kt` and `WelcomeFragment.kt` caused by me having an unexpected `.editorconfig` in  a parent directory.

File: app/src/main/java/mozilla/lockbox/action/DataStoreAction.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import mozilla.lockbox.flux.Action
+
+data class DataStoreAction(val type: Type) : Action {
+    enum class Type {
+        LOCK,
+        UNLOCK,
+        RESET,
+        SYNC
+    }
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/action/DataStoreAction.kt]: Are there any other types of action? I'm not _loving_ the prospect of 

```swift
dispatcher.dispatch(DataStoreAction.Type.LOCK)
```

File: app/src/main/java/mozilla/lockbox/store/DataStore.kt [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    data class State(
+        val status: Status,
+        val error: Throwable? = null
+    ) {
+        enum class Status {
+            UNPREPARED,
+            LOCKED,
+            UNLOCKED,
+            ERRORED
+        }
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/store/DataStore.kt]: Is this where SyncStatus will end up? If so, please could you comment.

File: app/src/main/java/mozilla/lockbox/store/DataStore.kt [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+ */
+class DataStore(
+    val dispatcher: Dispatcher = Dispatcher.shared,
+    val support: DataStoreSupport = FixedDataStoreSupport.shared
+) {
+    data class State(
+        val status: Status,
+        val error: Throwable? = null
+    ) {
+        enum class Status {
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/store/DataStore.kt]: Nit: Rename to match iOS.

**Comment by linuxwolf** [comment]
comments addressed IRL.  Last-minute test edition to verify list update on sync.

**Pull request merged**